src/dmd/e2ir.d

```diff
@@ -3520,15 +3520,18 @@ extern (C++) class ToElemVisitor : Visitor
                 e = addressElem(e, de.e1.type);
                 rtl = RTLSYM.DELMEMORY;
                 tb = (cast(TypePointer)tb).next.toBasetype();
+
+                /* `delete s` expressions were lowered to `_d_delstruct(s)` in
+                 * expressionsem.d. In order to still output `@safe`-ty, `nothrow`
+                 * and `nogc` errors related to dtors, the result was set to a
+                 * CommaExp made up of both the original and the lowered
+                 * expression. This is how e2ir.d gets here.
+                 * Since the lowering has already been made, the DeleteExp can
+                 * be ignored here.
+                 */
```

Comment:
How is this going to affect the traceGC hooks? If you look below, there is a call to `toTraceGC` which basically calls a wrapper of the hook if the `gc_profile` command line flag is passed. It seems that in this case, the wrapper call is ignored.